catalogs/catalog-messaging-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaCatalogOperations.java

```diff
@@ -92,39 +113,187 @@ public void initialize(Map<String, String> config, CatalogInfo info) throws Runt
                     e -> e.getKey().substring(CATALOG_BYPASS_PREFIX.length()),
                     Map.Entry::getValue));
     adminClientConfig.putAll(bypassConfigs);
-    adminClientConfig.put(BOOTSTRAP_SERVERS, config.get(BOOTSTRAP_SERVERS));
+    adminClientConfig.put(
+        AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, config.get(BOOTSTRAP_SERVERS));
     // use gravitino catalog id as the admin client id
-    adminClientConfig.put("client.id", config.get(ID_KEY));
+    adminClientConfig.put(AdminClientConfig.CLIENT_ID_CONFIG, config.get(ID_KEY));
 
     createDefaultSchema();
+    adminClient = AdminClient.create(adminClientConfig);
   }
 
   @Override
   public NameIdentifier[] listTopics(Namespace namespace) throws NoSuchSchemaException {
-    throw new UnsupportedOperationException();
+    NameIdentifier schemaIdent = NameIdentifier.of(namespace.levels());
+    if (!schemaExists(schemaIdent)) {
+      LOG.warn("Kafka catalog schema {} does not exist", schemaIdent);
+      throw new NoSuchSchemaException("Schema %s does not exist", schemaIdent);
+    }
+
+    try {
+      ListTopicsResult result = adminClient.listTopics();
+      Set<String> topicNames = result.names().get();
+      return topicNames.stream()
+          .map(name -> NameIdentifier.of(namespace, name))
+          .toArray(NameIdentifier[]::new);
+    } catch (InterruptedException | ExecutionException e) {
+      throw new RuntimeException("Failed to list topics under namespace " + namespace, e);
```

Comment:
Namespace is schema?

Reply:
yes, both are ok here I think

Reply:
If that，  the user needs to understand that namepsace is schema.

Reply:
fixed